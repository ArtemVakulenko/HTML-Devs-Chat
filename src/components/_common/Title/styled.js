import styled from 'styled-components';

export const AppTitle = styled.h1`
  font-size: 24px;
  margin: auto;
`;
