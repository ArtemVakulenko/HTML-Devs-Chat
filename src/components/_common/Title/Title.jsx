import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { AppTitle } from './styled';

const Title = ({ title, isTranslate = true }) => {
  const { t } = useTranslation();
  return (
    <AppTitle>
      {isTranslate ? t(title) : title}
    </AppTitle>
  );
};

export default Title;

Title.propTypes = {
  title: PropTypes.string.isRequired,
  isTranslate: PropTypes.bool,
};
