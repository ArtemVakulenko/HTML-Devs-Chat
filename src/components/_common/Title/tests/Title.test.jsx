import React from 'react';
import Title from '../Title';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);

describe('Title', () => {
  const props = {
    title: 'This Title',
  };
  it('Should match snapshot', () => {
    const component = global.shallow(<Title {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have h1', () => {
    const component = global.mount(<Title {...props} />);
    expect(component.find('h1')).toHaveLength(1);
  });
});
