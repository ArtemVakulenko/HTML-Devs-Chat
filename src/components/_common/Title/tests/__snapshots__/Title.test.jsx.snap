// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Should match snapshot 1`] = `"<h1 class=\\"styled__AppTitle-bMNsHC jWDbjD\\"></h1>"`;
